src/software/geom/rectangle.h

```diff
@@ -155,18 +155,18 @@ inline constexpr Rectangle::Rectangle(const Point &point1, const Point &point2)
 {
 }
 
-inline constexpr Rectangle::Rectangle(const Point &neg_x_neg_y_corner, double width,
-                                      double height)
-    : min_corner(neg_x_neg_y_corner), diagonal(width, height)
+inline constexpr Rectangle::Rectangle(const Point &sw_corner, double x_length, double y_length)
```

Comment:
The parameter names should stay `neg_x_neg_y_corner` etc as before, we don't want to use "north east south west"